resource_activity/tests/test_resource_activity.py

```diff
@@ -114,3 +114,93 @@ def test_activity_w_booked_resources(self):
         self.assertEquals(len(sale_order.order_line), 1)
         # not sure when tax is applied in test or not
         self.assertEquals(activity.sale_orders.amount_total, 100)
+
+
+    def test_update_resource_registration_date_end(self):
+        
+        date_start = datetime(2022, 2, 1, 1, 1, 1)
+        registration_1 = {
+            "attendee_id": self.partner_demo.id,
+            "quantity": 1,
+            "quantity_needed": 1,
+            "booking_type": "booked",
+            "resource_category": self.mtb_category.id,
+            "product_id": self.bike_product.id,
+        }
+
+        registration_2 = {
+            "attendee_id": self.partner_demo.id,
+            "quantity": 1,
+            "quantity_needed": 1,
+            "booking_type": "booked",
+            "resource_category": self.mtb_category.id,
+            "product_id": self.bike_product.id,
+        }
+
+        # Test that update_resource_registration_date_end udpates date_end
+        # when activity date_end is later
+        
+        date_end = date_start + timedelta(days=4)
+        date_done = date_start + timedelta(days=2)
+
+        activity_not_finished = self.env["resource.activity"].create(
+            {
+                "date_start": date_start,
+                "date_end": date_end,
+                # set by _onchange_allocation_start in real life
+                "resource_allocation_start": date_start,
+                # set by _onchange_allocation_end in real life
+                "resource_allocation_end": date_end,
+                "location_id": self.main_location.id,
+                "activity_type": self.activity_type.id,
+                "registrations": [(0, 0, registration_1), (0, 0, registration_2)],
+            }
+        )
+        activity_not_finished.search_all_resources()
+        activity_not_finished.reserve_needed_resource()
+        activity_not_finished.update_resource_registration_date_end(date_done)
```

Comment:
Shouldn't you rather close the activity and test that `set_activity_as_done` and `update_resource_registration_date_end` were called ?

Reply:
Yes you are right.